sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/YAMLToInternalMappers.java

```diff
@@ -539,45 +526,48 @@ private static PortsSpec from(
 
             if (rawPort.getVip() != null) {
                 final RawVip rawVip = rawPort.getVip();
-                final String protocol =
-                        StringUtils.isEmpty(rawVip.getProtocol()) ? DEFAULT_VIP_PROTOCOL : rawVip.getProtocol();
+                // Check that VIP names dont conflict with other port names. In practice this is only an issue when a
+                // custom prefix/name is defined for the VIP as uniqueness is already enforced for port names.
                 final String vipName = StringUtils.isEmpty(rawVip.getPrefix()) ? name : rawVip.getPrefix();
+                RawPort matchingRawPort = rawPorts.get(vipName);
+                if (matchingRawPort != null && matchingRawPort != rawPort) {
+                    throw new IllegalArgumentException(String.format(
+                            "Provided VIP prefix '%s' in port '%s' conflicts with other port also named '%s'. " +
+                            "Expected VIP prefix to not collide with other ports' names.",
+                            vipName, name, vipName));
+                }
+                //TODO(nickbp): Ideally we would also check here that multiple VIPs don't share names with each other.
+                //              However elastic currently does this. Not sure why.
```

Comment:
It does work, but at the moment it results in somewhat confusing endpoints output. I will actually remove this TODO as elastic's behavior (having all 'data' nodes show up under a 'data' VIP hostname) does make some sense.